readme.md

```diff
@@ -219,27 +220,28 @@ Then, go to [http://localhost:5000/](http://localhost:5000) and you should see a
   const { generate: generateId } = require('shortid')
   ```
 
-* **Your Answer:**
+* **Your Answer:** ShortId creates short non-sequential unique ids. The ^ code creates a shortid. The syntax is called 
+destructuring and in thsi case it matches objects.
 
 * **Question:** What routes are currently available to you as part of this application?
 
-* **Your Answer:**
+* **Your Answer:** /vegetables and /vegetables/:id and it also specifies if it is doesnt not match /vegtables, to return a 404 with `Could not find vegetable with ID of ${id}`.
 
 * **Question:** Look for `helpers.validate` in the `app.js` file. What is this and how does it work?
 
-* **Your Answer:**
+* **Your Answer:** This validates the object/variable you defined. In this case helpers is a imported module. The .validate() function returns an array of validation errors.
 
 * **Question:** Try creating a new vegetable. Then, try restarting your server. What happens to the data you posted and why?
 
-* **Your Answer:**
+* **Your Answer:** The data is posted in the body with an unique id. If you dont specify a name and a key, then you ill get 'Bad Request' because of the validate function.
 
 * **Question:** Take a look at the last two `app.use()` methods at the end of the file. Describe how both work and what `err` is refers to.
 
-* **Your Answer:**
+* **Your Answer:** The last app.use() is a error handler function that acts as a catch-all.
```

Comment:
Correct, the second to last app.use() will catch any unmatched requests and format an error.